tasks/ansible.yml

```diff
@@ -16,3 +16,13 @@
   command: ssh-keygen -b 2048 -t rsa -f /root/.ssh/id_rsa -q -N ""
   args:
     creates: /root/.ssh/id_rsa
+
+- name: Retrieve SSH public key
+  slurp:
+    src: /root/.ssh/id_rsa.pub
+  register: ansible_public_key_slurp
+
+- name: Save SSH public key to dummy host
```

Comment:
Hmm, the dummy host likely breaks other Ansible roles in unpredictable ways (iterating over all hosts is common).

Thoughts:

- There could be multiple Ansible hosts.
- Public key should be added as a fact for the current host.
- Maybe have a `openshift_ansible_host_group` variable which specifies the Ansible host group, that we can then iterate over?

Checking for empty `openshift_ssh_public_key` should be enough to fix a single node deployment.

Reply:
I've built it properly™️ now, without breaking compatibility of existing inventory files. Most importantly, I'm building the group you mentioned automatically from the existing `openshift_is_ansible_host` variable. Since additional Ansible hosts could also be regular Openshift nodes, I'm also adding the SSH key of all Ansible hosts to _all_ other nodes, including other Ansible hosts.